File: app/src/main/java/com/fastaccess/provider/markdown/extension/mention/MentionExtension.java
package com.fastaccess.provider.markdown.extension.mention;

import com.fastaccess.provider.markdown.extension.mention.internal.MentionDelimiterProcessor;
import com.fastaccess.provider.markdown.extension.mention.internal.MentionNodeRenderer;

import org.commonmark.Extension;
import org.commonmark.parser.Parser;
import org.commonmark.renderer.html.HtmlRenderer;

/**
 * Created by kosh on 20/08/2017.
 */

public class MentionExtension implements Parser.ParserExtension, HtmlRenderer.HtmlRendererExtension {
    private MentionExtension() {}

    public static Extension create() {
        return new MentionExtension();
    }

    @Override public void extend(Parser.Builder parserBuilder) {
        parserBuilder.customDelimiterProcessor(new MentionDelimiterProcessor());
    }

    @Override public void extend(HtmlRenderer.Builder rendererBuilder) {
        rendererBuilder.nodeRendererFactory(MentionNodeRenderer::new);
    }
}

File: app/src/main/java/com/fastaccess/github/platform/markwon/mention/MentionExtension.kt
package com.fastaccess.github.platform.markwon.mention


import com.fastaccess.github.platform.markwon.mention.internal.MentionDelimiterProcessor
import com.fastaccess.github.platform.markwon.mention.internal.MentionNodeRenderer
import org.commonmark.Extension
import org.commonmark.parser.Parser
import org.commonmark.renderer.html.HtmlNodeRendererFactory
import org.commonmark.renderer.html.HtmlRenderer

/**
 * Created by kosh on 20/08/2017.
 */

class MentionExtension private constructor() : Parser.ParserExtension, HtmlRenderer.HtmlRendererExtension {

    override fun extend(parserBuilder: Parser.Builder) {
        parserBuilder.customDelimiterProcessor(MentionDelimiterProcessor())
    }

    override fun extend(rendererBuilder: HtmlRenderer.Builder) {
        rendererBuilder.nodeRendererFactory(HtmlNodeRendererFactory { MentionNodeRenderer(it) })
    }

    companion object {

        fun create(): Extension {
            return MentionExtension()
        }
    }
}
